pkg/kn/commands/source/apiserver/create_flag.go

```diff
@@ -0,0 +1,77 @@
+// Copyright © 2019 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apiserver
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/spf13/cobra"
+	sources_v1alpha1 "knative.dev/eventing/pkg/apis/sources/v1alpha1"
+)
+
+const (
+	ApiVersionSplitChar           = ":"
+	ApiServerSourceArraySplitChar = ","
+	DefaultApiVersion             = "v1"
+)
+
+// ApiServerSourceUpdateFlags are flags for create and update a ApiServerSource
+type ApiServerSourceUpdateFlags struct {
+	ServiceAccountName string
+	Mode               string
+	Resources          string
+}
+
+// GetApiServerResourceArray is to return an array of ApiServerResource from a string. A sample is Event:v1:true,Pod:v2:false
+func (f *ApiServerSourceUpdateFlags) GetApiServerResourceArray() []sources_v1alpha1.ApiServerResource {
+	var resourceList []sources_v1alpha1.ApiServerResource
+	for _, r := range strings.Split(f.Resources, ApiServerSourceArraySplitChar) {
```

Comment:
I'm not sure if already had that discussion, but I would much prefer to allow multiple options for `--resource` instead of cramming everything into a single one, comma separated. It should be modelled much like `--traffic` on `kn service update`, to be consistent and and e.g. also for clarity.

Reply:
`--traffic` is `StringSliceVar`, for example `v1=50,v2=50`; I will follow `--traffic` to handle `--resource`.